nxc/modules/hyperv-host.py

```diff
@@ -0,0 +1,53 @@
+from impacket.dcerpc.v5.rpcrt import DCERPCException
+from impacket.dcerpc.v5 import rrp
+from impacket.examples.secretsdump import RemoteOperations
+
+
+class NXCModule:
+    name = "hyperv-host"
+    description = "Performs a registry query on the VM to lookup its HyperV Host"
+    supported_protocols = ["smb"]
+    opsec_safe = True
+    multiple_hosts = True
+
+    def __init__(self, context=None, module_options=None):
+        self.context = context
+        self.module_options = module_options
+  
+
+    def options(self, context, module_options):
+        """"""
+  
+      
+    def on_admin_login(self, context, connection):
+        self.context = context
+      
+        path = "SOFTWARE\Microsoft\Virtual Machine\Guest\Parameters"
+        key = "HostName"
+
+        remote_ops = RemoteOperations(connection.conn, False)
+        remote_ops.enableRegistry()
+
+        try:
+            ans = rrp.hOpenLocalMachine(remote_ops._RemoteOperations__rrp)
+            reg_handle = ans["phKey"]
+
+            ans = rrp.hBaseRegOpenKey(remote_ops._RemoteOperations__rrp, reg_handle, path)
+            key_handle = ans["phkResult"]
+
+            # Query
+            try:
+                data_type, reg_value = rrp.hBaseRegQueryValue(remote_ops._RemoteOperations__rrp, key_handle, key)
+                self.context.log.highlight(f"{key}: {reg_value}")
+            except Exception:
+                self.context.log.fail(f"Registry key {path}\\{key} does not exist")
+                return
+
+            rrp.hBaseRegCloseKey(remote_ops._RemoteOperations__rrp, key_handle)
+        except DCERPCException as e:
+            self.context.log.fail(f"DCERPC Error while querying registry: {e}")
```

Comment:
Also `ERROR_FILE_NOT_FOUND` should probably be just muted, as this is just the result of hyperv not being present

Reply:
@NeffIsBack tested the module on my lab against a Hyper-V VM and another non-VM and errors seem to be gone!

![image](https://github.com/user-attachments/assets/8daf7354-f5e9-46e5-badd-83f10b1b9216)

Reply:
@joaovarelas i changed it to only catch the `DCERPCException` to specifically only catch that error (the outer try&catch will catch normal exceptions) and added a debug statement, so we can track down what is happening with `--debug`, if needed some day.

If that looks alright to you we should be good to merge.

Reply:
@NeffIsBack thanks for clarifying, looks good to me! Regards